File: myautomaton/State.java
package myautomaton;

import java.util.HashMap;

public class State {

    boolean isFinal;
    boolean isDeathState;
    String name;
    HashMap<String, State> transitions;

    public State(boolean isFinal, boolean isDeathState,
            String name) {
        this.isFinal = isFinal;
        this.isDeathState = isDeathState;
        this.name = name;
        transitions = new HashMap<>();
    }

    public void addTranstition(String symbol, State state){
        transitions.put(symbol, state);
    }
    
    public State getNextState(String symbol){
        return transitions.get(symbol);
    }
    
    @Override
    public String toString() {
        return "State{" + "isFinal=" + isFinal + ", isDeathState=" + isDeathState + ", name=" + name + ", transitions=" + transitions.size() + '}';
    }
    
}

File: AutomatonPy/State.py
import collections
class State:
    def __init__(self, isFinalState, isDeathState, name):
        self.isFinalState = isFinalState
        self.isDeathState = isDeathState
        self.name = name
        self.transitions = collections.defaultdict(State)

    def addTransition(self, symbol, state):
        self.transitions[symbol] = state

    def getNextState(self, symbol):
        return self.transitions[symbol]